dev/com.ibm.ws.componenttest.2.0/src/componenttest/app/AssertionErrorSerializer.java

```diff
@@ -0,0 +1,216 @@
+    private static final String STACK_KEY = "stack";
+    private static final String FILE_NAME_KEY = "fileName";
+    private static final String LINE_NUMBER_KEY = "lineNumber";
+    //The FATServlet.doGet method name
+    private static final String DO_GET = "doGet";
+
+    /**
+     * Simplify an AssertionError that has been thrown from a test in a subclass of FATServlet.
+     * This will change the error message to include the FAT classname and test method name.
+     * It will also shorten the stacktrace to stop at FATServlet.doGet because everything after that
+     * will always be the same and is not of any interest.
+     *
+     * @param fatClass      The FAT class which originally threw the AssertionError
+     * @param fatMethodName The FAT method which originally threw the AssertionError
+     * @param e             The original AssertionError
+     * @return A simplified AssertionError
+     */
+    public static <T extends FATServlet> AssertionError simplify(Class<T> fatClass, String fatMethodName, AssertionError e) {
+        AssertionError assertionError = new AssertionError(fatClass.getSimpleName() + "." + fatMethodName + ": " + e.getMessage());
+        StackTraceElement[] originalStack = e.getStackTrace();
+        ArrayList<StackTraceElement> shortenedStack = new ArrayList<>();
+
+        for (StackTraceElement element : originalStack) {
+            String declaringClass = element.getClassName();
+            String methodName = element.getMethodName();
+            String fileName = element.getFileName();
+            int lineNumber = element.getLineNumber();
+            StackTraceElement newElement = new StackTraceElement(declaringClass, methodName, fileName, lineNumber);
+
+            shortenedStack.add(newElement);
+
+            //the stack beyond doGet is always the same so don't output any more
+            if ((FATServlet.class.getSimpleName() + ".java").equals(fileName) && DO_GET.equals(methodName)) {
+                break;
+            }
+        }
+
+        assertionError.setStackTrace(shortenedStack.toArray(new StackTraceElement[shortenedStack.size()]));
+        return assertionError;
+    }
+
+    /**
+     * Serialize out an AssertionError as a JSON String
+     *
+     * @param e The AssertionError
+     * @return A JSON String
+     */
+    public static String serialize(AssertionError e) {
+        JsonObject json = serializeAssertionError(e);
+        StringWriter writer = new StringWriter();
+        JsonWriter jsonWriter = Json.createWriter(writer);
+        jsonWriter.writeObject(json);
+        jsonWriter.close();
+        return writer.toString();
+    }
+
+    /**
+     * @param json A json string containing the serialized form of an AssertionError
+     * @return An instance of AssertionError
+     */
+    public static AssertionError deserialize(String json) {
+        StringReader reader = new StringReader(json);
+        JsonReader jsonReader = Json.createReader(reader);
```

Comment:
it is more efficient to use Json.createReaderFactory and store it statically and then make the Reader from the factory.  Same with createWriter and the other createXXX methods on Json.

Reply:
Done